### PR TITLE
configure evolutions base directory with evolutions.basePath

### PR DESCRIPTION
**Pull request by stephennancekivell**
Solution to this issue. https://github.com/playframework/playframework/issues/3966

### CODE REVIEW

File: framework/src/play-jdbc/src/main/scala/play/api/db/evolutions/Evolutions.scala [hunk 1]
```diff
@@ -79,10 +80,13 @@ case class DownScript(evolution: Evolution) extends Script {
  */
 object Evolutions {
 
+  val conf = ConfigFactory.load();
+  val evolutionsBasePath = conf.getString("evolutions.basePath", "evolutions")
```
Comment (jroper) [inline framework/src/play-jdbc/src/main/scala/play/api/db/evolutions/Evolutions.scala]: We shouldn't be reparsing the entire configuration statically just to get this configuration - the base path should be taken from an injection configuration object.

**Comment by jroper** [comment]
I think the best thing to do here is to remove these methods from the `Evolutions` object, they aren't helpful there, and rather just do the string building directly in `EnvironmentEvolutionsReader`, `ClassLoaderEvolutionsReader` etc.  The `EnvironmentEvolutionsReader` is what gets used in Play, it can also read the configuration from an injected configuration.

**Comment by jroper** [comment]
Actually, `EnvironmentEvolutionsReader` should depend on `EvolutionsConfig`, and this configuration should be added there, as you suggest in the issue.

As for configuration item names etc, it should be configurable globally and per database, and be something like this:

* The directory can be configured per datasource using `play.modules.evolutions.db.<datasourcename>.path`.  So, if `play.modules.evolutions.db.default.path = "foo"`, then evolutions are read from `foo/1.sql` etc.
* If it's not configured at the datasource level, that it defaults to `${play.modules.evolutions.path}/<datasourcename>`.  So, if `play.modules.evolutions.path = "bar"`, then evolutions for `default` are read from `bar/default/1.sql`.
* The default for `play.modules.evolutions.path` is `evolutions`.

I also think we should stop reading from the filesystem, just read straight from the classpath.

**Comment by stephennancekivell** [comment]
Thanks James, Ill look into it.

**Comment by stephennancekivell** [comment]
Im closing this issue, Im not going to finish it. We're using symlinks instead.

I did look into doing as you suggested James, but the rabbit hole was too deep.

**Comment by mkurz** [comment]
This has finally been implement in #11917

**Pull request closed without merging**